build.sbt

```diff
@@ -148,7 +152,40 @@ lazy val client = crossProject(JSPlatform, JVMPlatform)
   )
   .jsSettings(
     scalaJSModuleKind := ModuleKind.CommonJSModule,
-    artifactPath in (Compile, fullOptJS) := baseDirectory.value / "target" / "artifact" / "amf-client-module.js"
+    artifactPath in (Compile, fullOptJS) := baseDirectory.value / "target" / "artifact" / "amf-client-module.js",
+    moduleName := "amf-client-js",
+    customMappings := MappingFactory()
+      .map("ClientList").to("Array")
+      .map("ClientFuture").to("Promise")
+      .map("AmfCustomClass").to("AnotherCustomClass")
+      .map("ClientOption").to(option())
+      .map("ClientMap").to(dictionary())
+      .map("AnyVal").to("any")
+      .map("ClientLoader").to("ClientResourceLoader")
+      .map("ClientReference").to("ClientReferenceResolver")
+      .map("DocBuilder").to(overwrite("JsOutputBuilder"))
+      .map("Unit").to("void"),
+    namespaceReplacer := PrefixNamespaceReplacer("amf\\.client\\.", ""),
+    scalaFilteredClasses := ScalaClassFilterBuilder()
```

Comment:
Classes that we won't want to export in our typings file due to:
- use not exportable types in methods that are also used internally or that are being used client side thus breaking compatibility.
- types that declare namespaces that clash with one another.
- classes that declare methods/variables that class with each other.

this doesn't mean that they can be used from Javascript. They just won't be added to the typings declaration file.